loopy/check.py

```diff
@@ -668,27 +670,25 @@ def discard_dep_reqs_in_order(dep_reqs_to_vars, edges, order):
         :arg order: An instance of :class:`list` of instruction ids in which the
             *edges* graph is to be traversed.
         """
-        # predecessors: mapping from insn_id to its direct/indirect
-        # predecessors
-        predecessors = {}
-
-        for insn_id in order:
-            # insn_predecessors:insn_id's direct+indirect predecessors
-
-            # This set of predecessors is complete because we're
-            # traversing in topological order: No predecessor
-            # can occur after the instruction itself.
-            insn_predecessors = predecessors.pop(insn_id, set())
-
-            for pred in insn_predecessors:
-                dep_reqs_to_vars.pop(
-                    (insn_id, pred),
-                    # don't fail if pair doesn't exist
-                    None)
-
-            for successor in edges[insn_id]:
-                predecessors.setdefault(successor, set()).update(
-                        insn_predecessors | {insn_id})
+
+        dep_reqs_to_vars_dict = defaultdict(set)
```

Comment:
Alternate naming: `insn_to_req_deps`?, as unlike `dep_reqs_to_vars` it does store the variable which necessitated that dependency requirement.

Reply:
Done